src/server/load_statistics/linux.rs

```diff
@@ -56,29 +60,45 @@ impl ThreadLoadStatistics {
     /// the usage curve more smooth.
     pub fn record(&mut self, instant: Instant) {
         self.instants[self.cur_pos] = instant;
-        self.cpu_usages[self.cur_pos] = 0f64;
+        self.cpu_usages[self.cur_pos].clear();
         for tid in &self.tids {
             // TODO: if monitored threads exited and restarted then, we should update `self.tids`.
             if let Ok(stat) = thread::full_thread_stat(self.pid, *tid) {
-                self.cpu_usages[self.cur_pos] += thread::linux::cpu_total(&stat);
+                let total = thread::linux::cpu_total(&stat);
+                self.cpu_usages[self.cur_pos].insert(*tid, total);
             }
         }
         let current_instant = self.instants[self.cur_pos];
-        let current_cpu_usage = self.cpu_usages[self.cur_pos];
+        let current_cpu_usages = &self.cpu_usages[self.cur_pos];
 
         let next_pos = (self.cur_pos + 1) % self.slots;
         let earlist_instant = self.instants[next_pos];
-        let earlist_cpu_usage = self.cpu_usages[next_pos];
+        let earlist_cpu_usages = &self.cpu_usages[next_pos];
         self.cur_pos = next_pos;
 
         let millis = (current_instant - earlist_instant).as_millis() as usize;
+        let mut total_usage = 0;
         if millis > 0 {
-            let mut cpu_usage = calc_cpu_load(millis, earlist_cpu_usage, current_cpu_usage);
-            if cpu_usage > self.tids.len() * 100 {
-                cpu_usage = self.tids.len() * 100;
+            let mut loads = self.thread_loads.stats.lock();
+            for tid in current_cpu_usages.keys() {
+                loads.entry(*tid).or_default();
+            }
```

Comment:
Since `ThreadLoadStatistics::tids` won't be changed, we can determine what should store at `ThreadLoadPool::stats` in `ThreadLoadStatistics::new`, how about moving these codes to there?